src/main/java/com/gooddata/warehouse/WarehouseService.java

```diff
@@ -546,27 +549,35 @@ public WarehouseS3Credentials getWarehouseS3Credentials(final Warehouse warehous
         notNull(s3Credentials, "s3Credentials");
         notNull(s3Credentials.getUri(), "s3Credentials.links.self");
 
-        final WarehouseTask task = createWarehouseTask(s3Credentials.getUri(), s3Credentials, HttpMethod.DELETE);
+        final WarehouseTask task = createWarehouseTask(s3Credentials.getUri(), HttpMethod.DELETE, s3Credentials);
         return new PollResult<>(this, createS3PollHandler(s3Credentials.getUri(), task, Void.class, "delete"));
     }
 
     private WarehouseTask createWarehouseTask(final String targetUri,
-                                              final WarehouseS3Credentials s3Credentials,
                                               final HttpMethod httpMethod,
+                                              final WarehouseS3Credentials s3Credentials,
                                               final Object... args) {
-        final HttpEntity<WarehouseTask> taskHttpEntity;
+        final MappingJacksonValue jacksonValue = new MappingJacksonValue(s3Credentials);
+        jacksonValue.setSerializationView(WarehouseS3Credentials.UpdateView.class);
+
+        final HttpEntity<MappingJacksonValue> requestEntity = (httpMethod == HttpMethod.DELETE)
```

Comment:
This method seems to behave quite differently for DELETE - I suggest not to use the same private method as PUT and POST.

Reply:
extracted separate `createAdd/Update/DeleteTask` methods with different sets of arguments